sormas-ui/src/main/java/de/symeda/sormas/ui/task/TaskGridFilterForm.java

```diff
@@ -1,27 +1,71 @@
 package de.symeda.sormas.ui.task;
 
+import com.vaadin.v7.ui.ComboBox;
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.i18n.Descriptions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.region.RegionReferenceDto;
 import de.symeda.sormas.api.task.TaskCriteria;
-import de.symeda.sormas.api.task.TaskDto;
+import de.symeda.sormas.api.task.TaskIndexDto;
+import de.symeda.sormas.api.user.UserDto;
+import de.symeda.sormas.ui.UserProvider;
 import de.symeda.sormas.ui.utils.AbstractFilterForm;
 import de.symeda.sormas.ui.utils.FieldConfiguration;
+import de.symeda.sormas.ui.utils.FieldHelper;
 
 public class TaskGridFilterForm extends AbstractFilterForm<TaskCriteria> {
 
 	private static final long serialVersionUID = -8661345403078183133L;
 
 	protected TaskGridFilterForm() {
-		super(TaskCriteria.class, TaskDto.I18N_PREFIX);
+		super(TaskCriteria.class, TaskIndexDto.I18N_PREFIX);
 		getContent().removeComponent(APPLY_BUTTON_ID);
 	}
 
 	@Override
 	protected String[] getMainFilterLocators() {
 		return new String[] {
-			TaskDto.TASK_STATUS };
+			TaskIndexDto.TASK_CONTEXT,
+			TaskIndexDto.TASK_STATUS,
+			TaskIndexDto.REGION,
+			TaskIndexDto.DISTRICT,
+			TaskCriteria.CONTEXT_ENTITY_FREE_TEXT };
 	}
 
 	@Override
 	protected void addFields() {
-		addField(FieldConfiguration.pixelSized(TaskDto.TASK_STATUS, 200));
+		addField(FieldConfiguration.pixelSized(TaskIndexDto.TASK_CONTEXT, 140));
+		addField(FieldConfiguration.pixelSized(TaskIndexDto.TASK_STATUS, 140));
+
+		final UserDto user = UserProvider.getCurrent().getUser();
+		if (user.getRegion() == null) {
+			final ComboBox regionField = addField(FieldConfiguration.pixelSized(TaskIndexDto.REGION, 200));
+			regionField.addItems(FacadeProvider.getRegionFacade().getAllActiveAsReference());
+
+			final ComboBox districtField = addDistrictField();
+
+			regionField.addValueChangeListener(e -> {
+				RegionReferenceDto region = (RegionReferenceDto) e.getProperty().getValue();
+				FieldHelper
+					.updateItems(districtField, region != null ? FacadeProvider.getDistrictFacade().getAllActiveByRegion(region.getUuid()) : null);
+			});
+		} else {
+			final ComboBox districtField = addDistrictField();
```

Comment:
District filter should also only be displayed when the user does not have a district

Reply:
Done